service/sharddistributor/leader/namespace/manager.go

```diff
@@ -51,6 +51,10 @@ type ManagerParams struct {
 
 // NewManager creates a new namespace manager
 func NewManager(p ManagerParams) *Manager {
+	if !p.Cfg.Enabled {
+		return nil
```

Comment:
If returning nil here will require doing nil checks in other places, consider returning a no-op implementation

Reply:
There is a single place that is controlled by the same flag, so I think this should be enough.